crates/core/src/client/messages.rs

```diff
@@ -77,6 +78,29 @@ pub enum SerializableMessage {
     ProtocolUpdate(TransactionUpdateMessage<ProtocolDatabaseUpdate>),
 }
 
+impl SerializableMessage {
+    /// The number of rows in the payload
+    pub fn num_rows(&self) -> Option<usize> {
```

Comment:
I'd like to postpone this one if that's ok. Mainly because we're not abstracting over `NumRows`. I think once we have a need for abstracting over it, at that point we should add the trait.